docs/reference/module-overview.md

```diff
@@ -0,0 +1,284 @@
+
+The most common base class modules include:
+
+- `@apostrophecms/piece-type` to add a new custom piece type
+- `@apostrophecms/page-type` to add a custom page type
+- `@apostrophecms/piece-page-type` to add an index page and show pages for a piece type
+- `@apostrophecms/widget-type` to add a custom widget type
+
+```javascript
+// modules/custom-piece/index.js
+module.exports = {
+  extend: '@apostrophecms/piece-type',
+  // ...
+}
+```
+
+You can use any existing module as a base class for a custom module, including another custom module. To do so, make sure the extending module is instantiated _after_ its base class in `app.js`.
+
+In this example, the "Landing page" module uses the "Default page" as its base class. Both are custom module to this hypothetical website.
+
+```javascript
+// app.js
+require('apostrophe')({
+  modules: {
+    'default-page': {},
+    'landing-page': {}
+  }
+});
+```
+
+```javascript
+// modules/landing-page/index.js
+modules.export = {
+  extend: 'default-page'
+};
+```
+
+You should not use both `improve` and `extend` in a single module.
+
+### `improve`
+
+Similarly to `extend`, `improve` is used to name another existing module. Instead of _adopting_ the existing module's configuration, using `improve` _changes_ the named module's configuration.
+
+**This is only valid in modules that are built independently and then installed into Apostrophe apps**, either on their own or as a part of a bundle. It is most often used to improve core Apostrophe modules.
+<!-- TODO: link to a definition of a bundle when available. -->
+
+You might include this in a stand-alone module that adds functionality to the `@apostrophecms/image` core module:
+
+```javascript
+// index.js
+modules.export = {
+  improve: '@apostrophecms/image'
+  // Additional functionality ...
+};
+```
+
+You should not use both `improve` and `extend` in a single module.
+
+
+### `options`
+
+An object used to add additional, often optional, settings to a module. There are many options available depending on the module type. See the [module configuration options page](/reference/module-options.md) for more information.
+
+### `fields`
+
+[Doc type](/reference/glossary.md#doc) modules have some fields configured by default, such as the `title` and `slug` fields. The `fields` setting is used for additional field management.
+
+The `fields` object is configured with subsections: `add`, `remove`, and `group`.
+
+#### `add`
+
+An object of fields to add to the doc type's schema. See the [field type reference](/reference/field-types/README.md) for more on field type configuration.
+
+```javascript
+// modules/article/index.js
+modules.export = {
+  fields: {
+    add: {
+      subtitle: {
+        label: 'Subtitle',
+        type: 'string'
+      }
+    }
+  }
+};
+```
+
+#### `remove`
+
+An array of field names from the base class module to remove. Some core doc type fields cannot be removed since they required by core functionality.
```

Comment:
We should say which fields cant be removed, or at least give a short list. Rather than saying 'required by for functionality',  I think we could say. "Some fields, such as [x] and [y] are required and can't be removed.